File: TP01Q01 - Palíndromo em Java/Palindromo.java
class Palindromo{
       //metodo verifica se o metodo é Palindromo   
      public static boolean Palindromo(String s){
           boolean resp = true;
           int n = s.length();
 
           for(int i = n/2; i < n; i++ ){
              if(s.charAt(i) != s.charAt(n-1-i)){
                   resp = false;
             }
           }
              return resp;
     }

      public static boolean Fim(String s){
           return (s.length() == 3 && s.charAt(0) == 'F' && s.charAt(1) == 'I' && s.charAt(2) == 'M'); 
      }

       public static void main(String[] args){
           String[] s = new String[1000];
             int Nentrada = 0;           
  
               //chama metodo Fim verificando se é FIM            
             s[Nentrada] = MyIO.readLine();
           while((Fim(s[Nentrada]) == false)){
                 Nentrada++;
                s[Nentrada] = MyIO.readLine();
           }
              


            //Se for palindromo printa na tela SIM caso contrário printa NAO
              for (int i =0 ; i < Nentrada; i++){
                   if(Palindromo(s[i])){
                        MyIO.println("SIM");
                   } else {
                        MyIO.println("NAO");
              } 
               }
       }
}

File: TP01Q02 - Palíndromo em C/Palindromo.c
#include <stdbool.h>
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
       
        //Verifica se é Palindromo ou não
         bool Palindromo(char s[],int i){
          bool resp = true; 
          int p = strlen(s);

           for(int i = p/2; i < p; i++ ){
              if(s[i] != s[p-1-i]){
                   resp = false;
             }
           }
              return resp;
     }

         bool Fim(char h[]){
            return (strlen(h) == 3 && h[0] == 'F' && h[1] == 'I' && h[2] == 'M');
         }

       void main(){
           char s[1000][100];
           int o;
             
           //entrada verifica se a palavra é FIM     
           scanf(" %[^\n]s",s[o]);
          while(Fim(s[o]) == false){
              o++;
            scanf(" %[^\n]s",s[o]);
          }

        //saida printa na tela SIM se for Palindromo e NAO se não for
         for(int i = 0; i < o; i++){
             if(Palindromo(s[i],0) != true){
                 printf("NAO\n");
             }
              else{
                  printf("SIM\n");
               }
         }
   
        }